controllers/utils.go

```diff
@@ -53,16 +54,23 @@ func namespaceFromString(namespace string) v1.Namespace {
 	}
 }
 
-func filterNamespacesByNames(names []string, namespaceList []v1.Namespace) (namespaces []v1.Namespace) {
-	nameLookup := make(map[string]bool, len(names))
+func includeNamespacesByNames(patterns []string, namespaceList []v1.Namespace) (namespaces []v1.Namespace) {
+	var compiledPatterns []*regexp.Regexp
 
-	for _, name := range names {
-		nameLookup[name] = true
+	for _, pattern := range patterns {
+		rgx, err := regexp.Compile(pattern)
+		if err == nil {
```

Comment:
we should probably abort here or at least log the error?

Reply:
I thought about this too. At the moment it's at least documented in the property that invalid patterns will be ignored. I see several options:
* Abort, log the error and make in the status field the `ready=false` condition (and add another condition saying that validation failed)
* Create admission webhooks (https://sdk.operatorframework.io/docs/building-operators/golang/webhooks/) that can validate new or updated SyncConfig before storing them in etcd (actively prevent updating or creating a SyncConfig that has invalid syntax). Though very interesting to get experience with, it's also probably more work and adds complexity.